<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>591</width>
    <height>337</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Meal Tracker</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(56, 56, 63);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <item>
     <widget class="QGroupBox" name="formGroupBox">
      <property name="title">
       <string>Add / Edit Meal</string>
      </property>
      <layout class="QGridLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="labelName">
         <property name="text">
          <string>Meal Name:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="nameInput"/>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="labelCalories">
         <property name="text">
          <string>Calories:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QSpinBox" name="calorieInput">
         <property name="maximum">
          <number>2000</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="labelType">
         <property name="text">
          <string>Type:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="typeComboBox">
         <item>
          <property name="text">
           <string>Breakfast</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Lunch</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Dinner</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Snack</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="labelDateTime">
         <property name="text">
          <string>Date / Time:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QDateTimeEdit" name="dateTimeEdit">
         <property name="calendarPopup">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="2">
        <layout class="QHBoxLayout" name="buttonLayout">
         <item>
          <widget class="QPushButton" name="addButton">
           <property name="text">
            <string>Add Meal</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clearButton">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(126, 0, 2);</string>
           </property>
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTableWidget" name="mealTable">
      <column>
       <property name="text">
        <string>ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Calories</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Type</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Date/Time</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="actionButtons">
      <item>
       <widget class="QPushButton" name="updateButton">
        <property name="text">
         <string>Update</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="removeButton">
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="filterToggleButton">
        <property name="text">
         <string>Filters</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="undoButton">
        <property name="text">
         <string>Undo</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="redoButton">
        <property name="text">
         <string>Redo</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="filterGroupBox">
      <property name="visible">
       <bool>false</bool>
      </property>
      <property name="title">
       <string>Filter Meals</string>
      </property>
      <layout class="QGridLayout" name="filterLayout">
       <item row="0" column="0">
        <widget class="QCheckBox" name="filterNameCheckBox">
         <property name="text">
          <string>Name</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="filterNameInput">
         <property name="enabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QCheckBox" name="filterTypeCheckBox">
         <property name="text">
          <string>Type</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QComboBox" name="filterTypeCombo">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>Any</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Breakfast</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Lunch</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Dinner</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Snack</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="filterCaloriesCheckBox">
         <property name="text">
          <string>Calories</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="caloriesLayout">
         <item>
          <widget class="QLabel" name="minCalLabel">
           <property name="text">
            <string>Min:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="minCalInput">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="maximum">
            <number>2000</number>
           </property>
           <property name="minimumWidth">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="maxCalLabel">
           <property name="text">
            <string>Max:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="maxCalInput">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="maximum">
            <number>2000</number>
           </property>
           <property name="minimumWidth">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="2" colspan="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelLogic">
         <property name="text">
          <string>Logic:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="filterLogicCombo">
         <item>
          <property name="text">
           <string>AND</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>OR</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="applyFilterBtn">
         <property name="text">
          <string>Apply Filter</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="clearFilterBtn">
         <property name="text">
          <string>Clear Filter</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
